packages/vm/tests/BlockchainTestsRunner.js

```diff
@@ -95,25 +96,41 @@ module.exports = async function runBlockchainTest(options, testData, t) {
     }
   }
 
-  const numBlocks = testData.blocks.length
-  let currentBlock = 0
-  let lastBlock = false
+  let currentFork = common.hardfork()
+  let currentBlock
+  let lastBlock = 0
   for (const raw of testData.blocks) {
-    currentBlock++
-    lastBlock = (currentBlock == numBlocks)
+    lastBlock = currentBlock
     const paramFork = `expectException${options.forkConfigTestSuite}`
     // Two naming conventions in ethereum/tests to indicate "exception occurs on all HFs" semantics
     // Last checked: ethereumjs-testing v1.3.1 (2020-05-11)
     const paramAll1 = 'expectExceptionALL'
     const paramAll2 = 'expectException'
     const expectException = raw[paramFork] ? raw[paramFork] : raw[paramAll1] || raw[paramAll2] || raw.blockHeader == undefined
+
+    // here we convert the rlp to block only to extract the number
+    // we have to do this again later because the common might run on a new hardfork
     try {
+      let block = new Block(Buffer.from(raw.rlp.slice(2), 'hex'), {
+        common
+      }) 
+      currentBlock = bufferToInt(block.header.number)
+    } catch(e) {
+      handleError(e, expectException)
+      continue
+    }
+
+    if (currentBlock < lastBlock) {
+      // "re-org": rollback the blockchain to currentBlock (i.e. delete that block number in the blockchain plus the children)
+      t.fail("re-orgs are not supported by the test suite")
```

Comment:
Not completely getting this re-org scenario: is this actually triggered somewhere in the tests? Then some of the tests should fail according to the code here I would assume? Or is this a "just in case" implementation? 😛

Reply:
The problem is that if you disable this, most tests pass (see the skipped tests which have a few tests which are added), but not all of these blocks are evaluated (these are the re-org blocks). So the test is only executed partially. There is only one test which actually fails due to this. I just don't like that tests are only partially executed. One of the features of these tests seems to be that it does this reorg stuff, but we are not testing this, so I do not think we should just allow the tests and make them pass while we know that they are only partially executed.

It is rather cumbersome that these tests setup the canonical chain, then do some reorg, and then basically "reorg back" to the initial canonical chain, which means that the tip block of the chain in our implementation is also the tip block which is expected by the test suite, which is also the reason why these tests pass. The blocks in the reorg are not evaluated. If not clear let me know 😄  Related issue which tracks this problem: #879